spec/features/course/announcement_pagination_spec.rb

```diff
@@ -22,10 +23,19 @@
         visit course_announcements_path(course)
 
         expect(page).to have_selector('nav.pagination')
-        course.announcements.sorted_by_sticky.sorted_by_date.page(1).each do |announcement|
-          expect(page).to have_selector('div', text: announcement.title)
-          expect(page).to have_selector('div', text: announcement.content)
+        announcements = course.announcements.sorted_by_sticky.sorted_by_date.page(1)
+        announcements.each do |announcement|
+          expect(page).to have_selector('div.unread', text: announcement.title)
+          expect(page).to have_selector('div.unread', text: announcement.content)
         end
+
+        expect(course.announcements.unread_by(user).count).
+          to eq(announcement_count - announcements.size)
+
+        # Visit the page again should not reduce the unread count
```

Comment:
Since you provided an after_action, wouldn't it be better if we checked that unread count is 30 before the page is visited, and check that unread count is 0 after the page is loaded?